contrib/opencensus-ext-requests/opencensus/ext/requests/trace.py

```diff
@@ -74,22 +79,53 @@ def call(url, *args, **kwargs):
         if utils.disable_tracing_hostname(dest_url, blacklist_hostnames):
             return requests_func(url, *args, **kwargs)
 
+        path = parsed_url.path if parsed_url.path else '/'
+
         _tracer = execution_context.get_opencensus_tracer()
         _span = _tracer.start_span()
-        _span.name = '[requests]{}'.format(requests_func.__name__)
+        _span.name = '{}'.format(path)
         _span.span_kind = span_module.SpanKind.CLIENT
 
-        # Add the requests url to attributes
-        _tracer.add_attribute_to_current_span(HTTP_URL, url)
+        # Add the requests host to attributes
+        _tracer.add_attribute_to_current_span(
+            HTTP_HOST, dest_url)
 
-        result = requests_func(url, *args, **kwargs)
+        # Add the requests method to attributes
+        _tracer.add_attribute_to_current_span(
+            HTTP_METHOD, requests_func.__name__.upper())
 
-        # Add the status code to attributes
+        # Add the requests path to attributes
         _tracer.add_attribute_to_current_span(
-            HTTP_STATUS_CODE, str(result.status_code))
+            HTTP_PATH, path)
 
-        _tracer.end_span()
-        return result
+        # Add the requests url to attributes
+        _tracer.add_attribute_to_current_span(HTTP_URL, url)
+
+        try:
+            result = requests_func(url, *args, **kwargs)
+        except requests.Timeout:
+            _span.set_status(
+                status_module.Status(1, message='request timed out')
```

Comment:
I guess `DEADLINE_EXCEEDED` might be a better fit. Do we know where to get clarification for such mapping?

BTW, it'll be helpful to use a named const than a literal int for readability, I wonder if the grpc lib provides these constant variables already.

Reply:
It actually seems like a bug in the spec https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#client-errors-for-client-http-calls.

This is the gRPC error code https://grpc.github.io/grpc/core/md_doc_statuscodes.html.
Also in the server section `DEADLINE_EXCEEDED` does use 4 https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#mapping-from-http-status-codes-to-trace-status-codes.

Reply:
I agree with using named constants! My doubt was if we should maps internal `requests` exceptions to OpenCensus exceptions, which are then mapped to GRPC codes, or importe the GRPC codes into the contribs.

Reply:
Please take a look at the solution given!